### DISCUSSION TITLE
Is there any way to replace origin class instances with custom class?

### DISCUSSION THREAD

**Discussion by skies-unbound** [Minestom/Minestom#1390] [Help] [answered]
Is there any way or plan to add some utils to replace the origin classes with custom classes? I mean, does Minestom create a Player instance from `new Player()...` directly, or do we have some utils, such as EntityFactory to take over the works that allow developers to add more attributes to an origin class?

I want to find a way that can easily create some extra attributes (such as mana in RPG, attack range) for an individual unit. Is it possible to make Minestom call something like PlayerFactory#createPlayer to make a new instance rather than new an instance?

**Comment by mworzala** [comment, marked as answer]
`MinecraftServer.getConnectionManager().setPlayerProvider(...)`

https://javadoc.minestom.net/net/minestom/server/network/ConnectionManager.html#setPlayerProvider(net.minestom.server.network.PlayerProvider)
